Sources/ArcGISToolkit/Components/Offline/OfflineMapViewModel.swift

```diff
@@ -58,6 +58,11 @@ class OfflineMapViewModel: ObservableObject {
         guard case.success(let models) = preplannedMapModels,
               models.filter(\.status.isDownloaded).isEmpty
         else { return }
+        
+        // Verify that there are no downloaded on demand map areas.
+        guard let models = onDemandMapModels,
+              models.filter(\.status.isDownloaded).isEmpty else { return }
+                
```

Comment:
It seems that this code that checks if there are any areas left after the delete is essentially the same in both onRemove methods. It should be extracted out to a function and not duplicated.